### ISSUE TITLE
[Suggestion] DRYer clone/copy

### ISSUE THREAD

**Issue by dhritzkiv** [mrdoob/three.js#6981]
While noticing that `Group` didn't have `clone` (in r71 – I see it's fixed in dev), I had an idea for better default/inherited `clone` + `copy` methods.

The idea is to not require the explicit constructor to be called upon for each type of object. Instead:

```js
//in /src/core/Object3D.js

//currently (r72dev)
clone: function ( recursive ) {
	return new THREE.Object3D().copy( this, recursive );
}

//becomes
clone: function ( recursive ) {
	return new this.constructor().copy( this, recursive );
}
```

By calling `clone` on an instance of `Object3D`, the instance would automatically use the constructor of itself to create clones. E.g. `THREE.Group` would no longer need a `clone` method explicitly defined as it would inherit `Object3D`'s, and would work for any future derivatives of Object3D. I can't seem to think of any downsides of this (breaking of the inheritance chain, other behavioural changes).

Note: this doesn't only have to apply to `Object3D`, and can apply to `Material`, `Geometry`, etc.

If this is something that seems like a good idea, I'll gladly make the changes and submit a PR.

**Comment by dubejf** [comment]
Works for me - although you will need to be careful: the `recursive` parameters is not applied evenly and some entities like the geometries in `src/extras/` have non standard constructors and will need to keep their own `clone` functions.

What do you think @mrdoob @WestLangley ?

**Comment by mrdoob** [comment]
Sounds good to me. Definitely worth a try!

**Comment by dhritzkiv** [comment]
Closed with #6983
